### PR TITLE
Add haml-lint

### PR DESCRIPTION
**Pull request by davidrapson**
1. To confirm that the docker image changes in https://github.com/citizensadvice/design-system/pull/272 allow us to install and run other gems in CI
2. To ensure haml linting is consistent between projects

### CODE REVIEW

File: .haml-lint.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+  ClassesBeforeIds:
+    EnforcedStyle: 'id'
+
+  SpaceInsideHashAttributes:
+    enabled: false
+
+  RuboCop:
+    # @TODO: Probably turn this back on
+    enabled: false
```
Comment (davidrapson) [inline .haml-lint.yml:13]: This is also disabled in public-website. Enabling triggers a bunch of small warnings, worth fixing eventually but not work blocking a build for.

File: Gemfile.lock [hunk 2]
```diff
@@ -47,6 +47,14 @@ GEM
     haml (5.2.0)
       temple (>= 0.8.0)
       tilt
+    haml-lint (0.999.999)
```
Comment (luke-hill) [inline Gemfile.lock:50]: 😕

File: .haml-lint.yml (hunk 1 above)
Comment (luke-hill) [reply .haml-lint.yml:13]: Question) If this is being ran as a separate script. Does this need to be present here? Or will it auto-run it without it being present?

**Review by luke-hill — APPROVED** [review]
Seems legit

File: Gemfile.lock (hunk 2 above)
Comment (davidrapson) [reply Gemfile.lock:50]: Yeah kinda weird, but that's what gets added when running `bundle add haml-lint`.

File: .haml-lint.yml (hunk 1 above)
Comment (davidrapson) [reply .haml-lint.yml:13]: It auto-runs as part of haml-lint

**Pull request merged**